src/pkg/packager/actions.go

```diff
@@ -0,0 +1,165 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2021-Present The Zarf Authors
+
+// Package packager contains functions for interacting with, managing and deploying zarf packages
+package packager
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"regexp"
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/defenseunicorns/zarf/src/pkg/message"
+	"github.com/defenseunicorns/zarf/src/pkg/utils/exec"
+	"github.com/defenseunicorns/zarf/src/types"
+)
+
+// Run commands that a component has provided
+func (p *Packager) runComponentActions(actionSet types.ZarfComponentActionSet, actions []types.ZarfComponentAction) error {
+ACTION:
+	for _, a := range actions {
+		spinner := message.NewProgressSpinner("Running command \"%s\"", a.Cmd)
+		defer spinner.Success()
+
+		var (
+			ctx    context.Context
+			cancel context.CancelFunc
+			cmd    string
+			err    error
+		)
+
+		cfg := actionGetCfg(actionSet, a)
+
+		if cmd, err = actionCmdMutation(a.Cmd); err != nil {
+			spinner.Errorf(err, "Error mutating command: %s", cmd)
+		}
+
+		// Otherwise, run the command with a timeout handler
+		spinner.Updatef("Waiting for command \"%s\" (timeout: %d seconds)", cmd, cfg.MaxTotalSeconds)
```

Comment:
On slower machines this may briefly flash a 0 second timeout message, then update to no timeout (would be better to set the messages in the if IMO.